### ISSUE TITLE
API for Discord bots

### ISSUE THREAD

**Issue by ddugovic** [Infinite-Chess/infinitechess.org#259]
I maintain [Lichess Statbot](https://top.gg/bot/490949867657494530) which makes it easy to fetch and display player profiles, games, tournaments, blogs, etc. from Discord application slash commands.

Could Infinite Chess get an API similar to [Lichess API](https://lichess.org/api) to fetch such data, so I could make an Infinite Chess Statbot?

**Comment by Naviary2** [comment]
Interesting!

What would be the requirements you'd need from the API? There aren't too many features on the website right now. It could provide profile data (such as elo and join date), the lobby, and game count.

**Comment by ddugovic** [comment]
I think some of the more popular features are:
- Fetch current/recent game(s) by username
- Fetch rating leaderboard top 150 players (admittedly, this is more interesting if players have profiles e.g. FIDE title, Twitch/YouTube URL)
- Fetch daily puzzle URL
- Fetch current rating(s) & history by username

**Comment by Naviary2** [comment]
Yeah, there currently isn't a leaderboard, puzzles, or recent games built into the site...

I may add the ability to spectate games, at that time there'd be a way to pull current games but, not yet...
